### PR TITLE
Java Client API review cleanup

### PR DESCRIPTION
**Pull request by mikaelm12**
Hitting a couple of things from the API review here. Part of https://github.com/aspnet/SignalR/issues/2957

- ALL_CAPS Transfer Format enum
- Move HubConnection methods
- Make PingMessage a singleton
- Spotless clean up

### CODE REVIEW

**Comment by BrennanConroy** [comment]
LGTM

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java [hunk 1]
```diff
@@ -3,12 +3,23 @@
...
+{
+    private static PingMessage instance;
 
-    int type = HubMessageType.PING.value;
+    private PingMessage()
+    {
+    }
+
+    public static PingMessage getInstance() {
+        if (instance == null) {
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: This doesn't need to be lazy-initialized

Comment (anurse) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: Yeah, skip the lazy-init and just initialize it in the field.

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java [hunk 2]
```diff
@@ -3,12 +3,18 @@
...
+class PingMessage extends HubMessage
+{
+    private static PingMessage instance = new PingMessage();
 
-    int type = HubMessageType.PING.value;
+    private PingMessage()
+    {
+    }
+
+    public static PingMessage getInstance() {return instance;}
```
Comment (davidfowl) [inline clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: nit: formatting

Comment (mikaelm12) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: Do we always want a new line after a bracket? I'll add a spotless rule

Comment (BrennanConroy) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: I think it's more about the spaces here

Comment (mikaelm12) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/PingMessage.java]: Yeah I get that, but it's easier to normalize and write rules for if we just put newlines

**Pull request merged**